data/transactions/logic/eval.go

```diff
@@ -2549,15 +2699,29 @@ func (cx *EvalContext) globalFieldToValue(fs globalFieldSpec) (sv stackValue, er
 	case LatestTimestamp:
 		sv.Uint, err = cx.getLatestTimestamp()
 	case CurrentApplicationID:
-		sv.Uint, err = cx.getApplicationID()
+		sv.Uint = uint64(cx.appID)
 	case CurrentApplicationAddress:
-		var addr basics.Address
-		addr, err = cx.getApplicationAddress()
+		addr := cx.getApplicationAddress(cx.appID)
```

Comment:
should an error be generated if cx.appID is zero ?
( i.e. I can't think of any execution path that it *could* happen ), which is why I'm hesitant about this one.

Reply:
opGlobal prevents accessing an "app mode" global, which CurrentApplicationAddress is, unless it was run in app mode. And in app mode, the appID is set at the start of execution. I'll add a test that ensures that is true even for inners - that if you create an app with an inner transaction, when that app initializes, `cx.appID` is set properly.

Reply:
Added - so now we have more assurance that cx.appID is always set (non-zero) in app evaluation